clients/rospy/src/rospy/timer.py

```diff
@@ -210,7 +210,11 @@ def run(self):
         current_expected = rospy.rostime.get_rostime() + self._period
         last_expected, last_real, last_duration = None, None, None
         while not rospy.core.is_shutdown() and not self._shutdown:
-            r.sleep()
+            try:
+                r.sleep()
+            except rospy.exceptions.ROSInterruptException as e:
+                if rospy.core.is_shutdown():
+                    break
```

Comment:
Before the update there was a `raise` line in here. My previous comment only referred to the `else` line. I think the `raise` still needs to stay otherwise this function changes behavior significantly.